middlewares/studentAssessmentAccess.js

```diff
@@ -0,0 +1,168 @@
+//var ERR = require('async-stacktrace');
+//var _ = require('lodash');
+//var sha256 = require('crypto-js/sha256');
+const express = require('express');
+var router = express.Router();
+
+var logger = require('../lib/logger');
+var csrf = require('../lib/csrf');
+var config = require('../lib/config');
+
+var timeout = 24; // hours
+
+router.get('/', function(req, res, next) {
+
+// Future feature
+//    // SEB: check BrowserExamKey
+//    if ('x-safeexambrowser-requesthash' in req.headers) {
+//        checkBrowserExamKeys(res, req.headers['x-safeexambrowser-requesthash']);
+//    }
+
+    // SEB: check user-agent
+    if ('user-agent' in req.headers) {
+        checkUserAgent(res, req.headers['user-agent']);
+    }
+
+// Future feature
+//    // Course Instance view (only show allowed exam)
+//    if (!('assessment' in res.locals)) {
+//        //console.log('In Course Instance Page');
+//        //res.locals.authz_data.mode = 'SEB';
+//        //console.log(res.locals);
+//    }
+
+    // SafeExamBrowser protect the assesment
+    if ('authz_result' in res.locals &&
+        res.locals.authz_result.mode == 'SEB') {
+
+        // If the assessment is complete, use this middleware to show the logout page
+        if ('assessment_instance' in res.locals && res.locals.assessment_instance.open == false) {
+            return badSEB(req, res);
+        }
+
+        // If any of our auth checks didn't pass, fail (send to download)
+        if (res.locals.authz_data.mode != 'SEB') {
```

Comment:
We already have `res.locals.authz_result.mode == 'SEB'` to get to this point, so I think this code can't happen.

Reply:
`authz_data` vs `authz_result`. I think `authz_result` is what the assessment requires, `authz_data` reflects our current emulation.

Reply:
Excellent point. Who made up these ridiculously confusing names?!? :-)